tests/testsuite/cargo_command.rs

```diff
@@ -341,6 +341,58 @@ fn cargo_subcommand_env() {
         .run();
 }
 
+#[cfg(unix)]
+#[cargo_test]
+fn cargo_cmd_bin_path_env_interaction() {
+    use std::io::Write;
+    use std::os::unix::fs::PermissionsExt;
+
+    let root = paths::root();
+    let bin = root.join("bin");
+    fs::create_dir(&bin).unwrap();
+    fs::File::create(&bin.join("cargo-foo"))
+        .unwrap()
+        .write_all(
+            br#"!/usr/bin/env bash
+                echo INSIDE
+            "#,
+        )
+        .unwrap();
+
+    fs::File::create(&root.join("cargo-foo"))
+        .unwrap()
+        .write_all(
+            br#"!/usr/bin/env bash
+                echo OUTSIDE
+            "#,
+        )
+        .unwrap();
+
+    fs::set_permissions(root.join("cargo-foo"), fs::Permissions::from_mode(0o555)).unwrap();
+    fs::set_permissions(bin.join("cargo-foo"), fs::Permissions::from_mode(0o555)).unwrap();
+
+    cargo_process("foo")
+        .env("CARGO_HOME", &root)
+        .with_stdout_contains("INSIDE")
+        .run();
+
+    cargo_process("foo")
+        .env("PATH", format!("{}:{}", bin.display(), root.display()))
+        .env("CARGO_HOME", &root)
+        .with_stdout_contains("INSIDE")
+        .run();
+
+    println!("{}", root.display());
+    println!("{}", bin.display());
+    println!("{}", format!("{}:{}", root.display(), bin.display()));
+    cargo_process("foo")
+        .env("PATH", format!("{}:{}", root.display(), bin.display()))
+        .env("CARGO_HOME", &root)
```

Comment:
`cargo-test-support` is creating a `CARGO_HOME` for you, could you use that instead of creating your own and setting `CARGO_HOME` explicitly?

Reply:
Done. I wasn't aware how it all works.